cc_plugin_ugrid/checker_10.py

```diff
@@ -166,5 +215,199 @@ def check_connectivity(self, ds):
+        nEdges respectively"
+
+        examples
+        --------
+        face_coordinates = 'Mesh2_face_x Mesh2_face_y'
+        """
+
+        level = BaseCheck.MEDIUM
+        score = 0
+        out_of = 1
+        messages = []
+        desc = 'Optional attribute pointing to the the auxiliary coordinate' + \
+                ' variables associated with the location of faces and edges'
+
+        dep = self.check_connectivity(ds)
+        if not dep or dep.value[0] != dep.value[1]:
+            # Skip if no connectivity
+            return None
+        dep = self.check_face_dimension(ds)
+        if not dep or dep.value[0] != dep.value[1]:
+            # Skip if no face dimension
+            return None
+
+        try:
+            mesh = ds.get_variables_by_attributes(cf_role='mesh_topology')[0]
+            if mesh.topology_dimension >= 2:
+                face_coords = mesh.face_coordinates
+                # split and get first element, accessed through variables
+                mesh_faces = ds.dimensions[ds.variables[face_coords.split(' ')[0]].dimensions[0]]
+                assert int(mesh_faces.size) == self.face_dim.size
+                score +=1
+            else:
+                # skip test because a 1-D array won't have face_coordinates
+                return None
+        except AttributeError:
+            # face_coordinates not in mesh; passes check because it's optional
+            mesh_faces = None
+            score += 1
+        except AssertionError:
+            # mesh_faces != nfaces
+            m = 'It seems that the face_coordinates' + \
+                ' attribute does not match face_dimension.'
+            messages.append(m)
+        if mesh_faces:
+            try:
+                # does mesh_faces equal the OTHER value of the conn. dims?
+                assert int(mesh_faces.size) == ds.variables[self.connectivity].shape[0]
+                m = 'face_coordinates seems to match the incorrect dimension' + \
+                'of _connectivity.'
+                messages.append(m)
+            except AssertionError:
+                # mesh_faces does not match either dimension of face_dimension
+                # face_coordinates should fail
+                # do I make face_dimension fail from here?
+                m = 'Your face_coordinates does not match either' + \
+                    'dimension of your face_dimension attribute.'
+                messages.append(m)
+
+        return self.make_result(level, score, out_of, desc, messages)
+
+
+    def check_location_mesh_in_variables(self, ds):
+        """
+        Loops through each variable to see if its dimensions contain time,
+        nodes, or faces. If so, imlpements a check for a 'location'
+        attribute in the variable; fails if it does not exist.
+        """
+
+        # DEPENDENCIES
+        self.check_face_dimension(ds)
+        self.check_node_coordinates_exist(ds)
+        if ds.get_variables_by_attributes(standard_name='time'):
```

Comment:
I don't think we need time here, there's no requirement a UGRID variable be time-varying. I may have led you down wrong path here. 🤕